packages/datadog-plugin-jest/src/index.js

```diff
@@ -40,9 +40,18 @@ function createWrapTeardown (tracer, instrumenter) {
 }
 
 let params = {}
+const spanByName = {}
 
 function createHandleTestEvent (tracer, testEnvironmentMetadata, instrumenter) {
   return async function handleTestEventWithTrace (event) {
+    if (event.name === 'test_fn_failure') {
+      const context = this.getVmContext()
+      if (context) {
+        const { currentTestName } = context.expect.getState()
+        const testSpan = spanByName[currentTestName]
```

Comment:
we can't read directly from the scope here through `tracer.scope().active()` because it is not within the spec function, so we need to store spans this way

Reply:
Can this be stored elsewhere? It happens often that the scope doesn't work within an integration and we store the value on something else. It can be pretty much anything that allows grabbing the span later on, some sort of context object, or the test object itself, something like that.

The main issue with storing this in this way is that it's prone to memory leak. Of course this integration being for a short-lived process, it might be fine, but let's make sure there is absolutely no other option before going this route.

Reply:
yeah I think in this case it will be easier that in the parameterized tests' case. I'll do it 👍

Reply:
done in https://github.com/DataDog/dd-trace-js/pull/1361/commits/65127376830e41200df92a5d92fb3d6d902d9635 👍